plugins/SegmentEditor/javascripts/Segmentation.js

```diff
@@ -760,16 +760,15 @@ Segmentation = (function($) {
             var segmentStr = self.currentSegmentStr;
             var encSegment = jQuery(jQuery('.segmentEditorPanel').get(0)).data('uiControlObject').uriEncodeSegmentDefinition(segmentStr);
 
-            var url = window.location.href;
-            //  URL might have format index.php?aparam=avalue#?anotherparam=anothervalue
-            // Need to strip off stuff before second ? as it mucks with updateParamValue
-            url = url.replace(/\?[\S]*\?/, '?');
-            // Show user the Visits Log so that they can easily refine their new segment if needed
-            url = broadcast.updateParamValue('viewDataTable=VisitorLog', url);
-            url = broadcast.updateParamValue('module=Live', url);
-            url = broadcast.updateParamValue('action=getLastVisitsDetails', url);
-            url = broadcast.updateParamValue('segment=' + encSegment, url);
-            url = broadcast.updateParamValue('inPopover=1', url);
+            var url = $.param({
+                date: piwik.currentDateString,
+                period: piwik.period,
+                idSite: piwik.idSite,
+                module: 'Live',
+                action: 'getLastVisitsDetails',
+                segment: encSegment,
+                inPopover: 1,
+            });
```

Comment:
That one had been a bit too complex before. The URL for the visits log doesn't need to be built of the current url hash. Doing that actually caused comparison parameters and other unneeded parameters to be part of the loaded url. And in case the url or hash contained an unexpected count of `?` this might even have failed and an incorrect url was loaded.